osmnx/geometries.py

```diff
@@ -351,43 +349,46 @@ def _create_gdf(response_jsons, polygon, tags):
 
         # extract geometries from the downloaded osm data
         for response_json in response_jsons:
-            # Parses the JSON of OSM nodes, ways and (multipolygon) relations to dictionaries of
-            # coordinates, Shapely Points, LineStrings, Polygons and MultiPolygons
+            # Parses the JSON of OSM nodes, ways and (multipolygon) relations
+            # to dictionaries of coordinates, Shapely Points, LineStrings,
+            # Polygons and MultiPolygons
             for element in response_json["elements"]:
 
                 # id numbers are only unique within element types
                 # create unique id from combination of type and id
                 unique_id = f"{element['type']}/{element['id']}"
 
-                # add elements that are not nodes and that are without tags or with empty tags
-                # to the untagged_element_ids set (untagged nodes are not added to the geometries
-                # dictionary at all)
-                if (element["type"] != "node") & (("tags" not in element) or (not element["tags"])):
+                # add elements that are not nodes and that are without tags or
+                # with empty tags to the untagged_element_ids set (untagged
+                # nodes are not added to the geometries dict at all)
+                if (element["type"] != "node") and (
+                    ("tags" not in element) or (len(element["tags"]) == 0)
+                ):
                     untagged_element_ids.add(unique_id)
 
                 if element["type"] == "node":
                     # Parse all nodes to coords
-                    coord = _parse_node_to_coord(element=element)
-                    coords[element["id"]] = coord
-                    # If the node has tags and the tags are not empty parse it to a Point
-                    # Empty check is necessary for JSONs created from XML where nodes without tags
-                    # are assigned tags = {}
-                    if "tags" in element and element["tags"]:
+                    coords[element["id"]] = _parse_node_to_coords(element=element)
+
+                    # If the node has tags and the tags are not empty parse
+                    # it to a Point. Empty check is necessary for JSONs created
+                    # from XML where nodes without tags are assigned tags = {}
+                    if "tags" in element and len(element["tags"]) > 0:
                         point = _parse_node_to_point(element=element)
                         geometries[unique_id] = point
 
                 elif element["type"] == "way":
                     # Parse all ways to linestrings or polygons
                     linestring_or_polygon = _parse_way_to_linestring_or_polygon(
-                        element=element, coords=coords, polygon_features=polygon_features,
+                        element=element, coords=coords
                     )
                     if linestring_or_polygon:
```

Comment:
@AtelierLibre What are we checking here? `linestring_or_polygon` is a dict right? Are we checking if it's empty (or, alternatively, if it's null)? If so, let's make that explicit like `len(linestring_or_polygon) > 0`.

Reply:
@gboeing I started `geometries` by duplicating `pois`. This is a remnant of [line 344](https://github.com/gboeing/osmnx/blob/f05b359c418ff0139f6696d504b766a35775ef7c/osmnx/pois.py#L344). I imagine it was there to catch the possibility that [_parse_polygonal_poi()](https://github.com/gboeing/osmnx/blob/f05b359c418ff0139f6696d504b766a35775ef7c/osmnx/pois.py#L142) wouldn't return anything because it triggered the exception due to an invalid geometry. You're right that the new [_parse_way_to_linestring_or_polygon()](https://github.com/gboeing/osmnx/blob/998613e249e8957510aa6abb091161604fb03104/osmnx/geometries.py#L493) doesn't behave like that and is intended to catch the specific errors, log them and return an emtpy geometry as part of a dictionary that is always created. I think this is redundant and can be removed.

Reply:
Resolved in 96a4489